app/components/site-type-select.js

```diff
@@ -0,0 +1,22 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+export default class SiteTypeSelectComponent extends Component {
+  @service fastboot;
+  @service store;
+  siteTypes;
+
+  constructor(...args) {
+    super(...args);
+
+    if (!this.fastboot.isFastBoot) {
+      this.siteTypes = this.loadSiteTypesTask.perform();
+    }
+  }
+
+  @task
+  *loadSiteTypesTask() {
+    return yield this.store.findAll('site-type');
```

Comment:
I would pass the [`reload: true`](https://api.emberjs.com/ember-data/3.27/classes/Store/methods/query?anchor=findAll) option to findAll. By default store.findAll returns records from the store if some are loaded already. This sounds nice on paper, but in practise that results in weird behavior:

Let's say the user loads the app on a specific site details page. That details page shows the site type relationship so the store will have 1 record of the `site-type` model. If the user then goes to the edit page, this call will instantly resolve to the 1 record of the store (and it will do a fetch in the background to find all of them).

So in that scenario the select will have only 1 option until the background request completes, which looks like a bug to end users.

Passing reload forces the fetch to be resolved before returning any data.

Reply:
Thanks for the explanation! 😄